go/internal/logutil/tyber.go

```diff
@@ -0,0 +1,39 @@
+package logutil
+
+import (
+	"net"
+	"strings"
+
+	"berty.tech/berty/v2/go/pkg/errcode"
+)
+
+type TyberLogger struct {
+	host string
+	conn net.Conn
+}
+
+func NewTyberLogger(host string) (*TyberLogger, error) {
+	// FIXME: support optional flags
+	//         --tyber-host=1.2.3.4:4242,ignore-connection-refused
+	//         --tyber-host=1.2.3.4:4242,no-reconnect
+	host = strings.TrimSpace(host)
+	if !strings.Contains(host, ":") {
+		host += ":4242"
+	}
+	logger := TyberLogger{host: host}
+	var err error
+	logger.conn, err = net.Dial("tcp", host)
+	if err != nil {
+		return nil, errcode.TODO.Wrap(err)
+	}
+	return &logger, nil
```

Comment:
use a context here
```suggestion
func NewTyberLogger(ctx context.Context, host string) (*TyberLogger, error) {
	host = strings.TrimSpace(host)
	if !strings.Contains(host, ":") {
		host += ":4242"
	}

	var d net.Dialer
	conn, err := d.DialContext(ctx, "tcp", host)
	if err != nil {
		return nil, errcode.TODO.Wrap(err)
	}

	return &TyberLogger{
		conn: conn,
		host: host,
	}, nil
}

```

Reply:
Wouldn't this add 2 sec to the startup time of the app whenever tyber is not available ? (blocking)

Reply:
It will have the same behavior as `Dial` method, except this: 
> If the context expires before the connection is complete, an error is returned. Once successfully connected, any expiration of the context will not affect the connection. `

Reply:
yes so it waits for the connection to complete or 2 seconds, you remove the goroutine that does that in parallel so it indeed adds 2 sec of boot time I think

Reply:
This will be refactored in [this PR](https://github.com/berty/berty/pull/3298) (see point 6) anyway, so it will no be a problem anymore to dial Tyber few minutes after the start of the session.
We will be able to handle more usecase:
- Tyber host address is updated in app settings
- mDNS detected a new Tyber host on the network, display a popup on the app to ask user if he want to forward logs